src/plugins/simulator/visualizations/qt-opengl/qtopengl_camera.cpp

```diff
@@ -23,41 +23,38 @@ namespace argos {
       /* Get positional attributes */
       GetNodeAttribute(t_tree, "position", Position);
       GetNodeAttribute(t_tree, "look_at", Target);
-      /* Calculate the Forward vector */
-      Forward = (Target - Position).Normalize();
       /* Was the up vector specified? */
       if(!NodeAttributeExists(t_tree, "up")) {
          /* Calculate the Left vector
             It is located on a plane parallel to XY
             It is perpendicular to the projection of Forward on that plane
          */
-         if(Forward.GetX() != 0 || Forward.GetY() != 0) {
-            CVector2 cLeftXY(Forward.GetX(), Forward.GetY());
+         CVector3 cForward((Target - Position).Normalize());
+         CVector3 cLeft;
```

Comment:
Why did you remove the `Left` vector from the struct and recalculated it here privately? Having the `Left` vector available is a good idea for programmatically changing the camera code in the `QtUserFunctions`, in case people would like to have that possibility.

Reply:
The problem is related to the interpolation. It doesn't make sense to constantly recalculate these vectors unless the user want to move the camera via user input

Reply:
I don't think I understand :( Having a `Left` vector in the struct is the same in terms of when to compute it, but it makes it much simpler and faster to rotate around that vector instead of having to recalculate it.

Reply:
So we now have a `m_sActivePlacement` that represents the current placement of the camera (including in between two placements from the array `m_arrPlacements` during interpolation). It is true, that we could precalculate several things inside `SPlacement` which would give a minor performance boost while rotating or moving the camera with the mouse. On the other hand, we now need to recalculate these vectors on each time step for `m_sActivePlacement` during interpolation to keep it in a correct state even though we are probably not going to use them. Removing this additional data from `SPlacement` may also bring a minor performance increase since the overall size of `m_arrPlacements` is smaller.